lib/cronofy/auth.rb

```diff
@@ -3,18 +3,19 @@
 module Cronofy
   class Auth
     API_URL = 'https://api.cronofy.com'
-    APP_URL = 'https://app.cronofy.com'
 
-    def initialize(client_id, client_secret, token)
+    attr_reader :access_token
+
+    def initialize(client_id, client_secret, token, refresh_token=nil)
       @client = OAuth2::Client.new(client_id, client_secret, site: API_URL)
-      @token = token
+      @access_token = OAuth2::AccessToken.new(@client, token, { refresh_token: refresh_token })
     end
 
     def user_auth_link(redirect_uri)
       url = @client.auth_code.authorize_url(
         :redirect_uri => redirect_uri,
         :response_type => 'code'
-      ).gsub(API_URL, APP_URL)
+      )
```

Comment:
I don't think `gsub` is the best way to do this but this needs to generate an `app.cronofy.com` URL. Also, the `scope` parameter should have the spaces escaped as `+` or `%20` so it's properly URI encoded.

Perhaps the best way is to create two clients in the constructor: `@api_client` and `@auth_client`. The former being used for `@access_token` and therefore API calls, the latter for generating this URL and perhaps the `#get_token_from_code`?